### PR TITLE
Product page v2 accessible in menu

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | To ease the starting work on automated tests and QA for product page v2 a link is added in the menu to access the new list page directly Also the feature flag for products is now ignored meaning the page is always enabled even if the product flag is not This PR does NOT clean the feature flag code, this will have to be done in another PR
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | ~
| How to test?      | Make a fresh install, the menu should have a new entry after "Products" which is "Products (experimental)" The experimental page is now always accessible regardless of the feature flag values along with all buttons related to it
| Possible impacts? | ~

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27481)

### CODE REVIEW

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
If you remove the feature flag, please remove `FEATURE_FLAG_PRODUCT_PAGE_V2` const too

**Comment by jolelievre** [comment]
@PierreRambaud as I explained in the description I didn't remove the feature flag per say, I just ignore its value in this PR

The removal of the feature flag itself is planned in another PR, because we need to think about how we handle it since it's the only available feature flag Should we keep the form empty, should we remove the controller? hide it? Do we keep the const values for backyard compatibility or is it ok to remove them? Do we clean the associated wording?

All those questions are part of how we handle the feature flag feature, it's not the scope of this PR which is simply to ease the work of QA team and anticipate the fact that the page will not be flaggable at some point.

So unless we answer all the questions about feature flag in the coming hours I'd like to at least move on this PR so that QA automation can be started early.

**Review by PierreRambaud — DISMISSED** [review]

**Review by saulaski — DISMISSED** [review]
LGTM

**Review by PierreRambaud — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by matthieu-rolland** [comment]
I removed the `waiting for wording` label since those are not new wordings

**Review by HanaRebaiQA — COMMENTED** [review]
Hello @jolelievre 

I have tested this PR and i detected theses abnormal behavior : 


1. The path to the Products (Standard) page is changed to (Categories > Products (experimental) ) ==> It can be risk of down of Automated Tests after merging the PR
2. Both menus (Products et Products V2) are visible in white when we are in a single page
3. The Product v2 menu is not accessible following an upgrade from 1783 to develop with PR
	I install this PR https://github.com/PrestaShop/autoupgrade/pull/434 in 1783 to do the upgrade to develop with the current PR
it is not an annoying bug if the menu is to be redesigned for the release of V8 (replacement of Product by Product V2)

Could you please check and feedback?

Thanks!

**Review by HanaRebaiQA — CHANGES_REQUESTED** [review]
Hello @jolelievre

I have tested this PR and i detected theses abnormal behavior :

1. The path to the Products (Standard) page is changed to (`Categories > Products (experimental)` ) ==> It can be risk of down of Automated Tests after merging the PR
image
image

2. Both menus (Products et Products V2) are visible in `white` when we are in a single page
image
image

3. The Product v2 menu is not accessible following an upgrade from 1783 to develop with PR

I install this [PR](https://github.com/PrestaShop/autoupgrade/pull/434) in **1783** to do the upgrade to **develop with the current PR.**

https://user-images.githubusercontent.com/92912932/152814090-ff3b7158-a418-41ed-9bc7-e38113781b00.mp4


==> It is not an annoying bug if the menu is to be redesigned for the release of V8 (replacement of Product by Product V2)
Could you please check and feedback?

Thanks!

**Comment by jolelievre** [comment]
Thanks @HanaRebaiQA 

indeed the fact that we have two link in the menu pointing on the same legacy controller (AdminProducts) is bound to create some irregular behaviours (because the menu doesn't expect such use cases). The problem of autoupgrade is not relevant because this behaviour is only temporary, before the version is released the old link will be removed and only one will be present which will correct all inconsistencies.

So the only question remaining is about automated tests, since this PR aims to ease the life of both QA and automated tests it's not up to me to decide this. So this should be decided by @boubkerbribri and @Robin-Fischer-PS I guess?

**Comment by Robin-Fischer-PS** [comment]
@jolelievre Could it be possible to display `Categories > Products` instead of `Categories > Products (experimental)` in the path (for both V1 and V2 pages) ? It would resolve the remaining issue that might indeed cause some troubles to automated tests.

**Comment by boubkerbribri** [comment]
I don't think the wording and 2 active tabs can have a impact on automated tests.
Technically, we will check that :
- `#Products_v2` is active.
-  Cheking the title of the page not the path after accessing the page. But even if we choose the path, the wording to check can change later.

**Comment by jolelievre** [comment]
@Robin-Fischer-PS the breadcrumb is automatically built based on the menu content it seems like it uses the last matching element which is why it showed the experimental link

I don't want to modify anything in the breadcrumb code so I just switched the two links in the content, so now `Products (experimental)` is displayed first in the menu list but this way the breadcrumb uses the same link `Products` which points to the legacy page

However, this seems to be breaking all the sanity tests now, probably because the UI tests use the first link matching `AdminProducts` (@boubkerbribri will probably have more info on this)

So TLDR; we need to choose which solution has the least impact but in any case no modification will be done on the menu cr breadcrumb code as the purpose here is simply to add a temporary link during development and testing phase

**Comment by boubkerbribri** [comment]
> However, this seems to be breaking all the sanity tests now, probably because the UI tests use the first link matching `AdminProducts` (@boubkerbribri will probably have more info on this)

Can we make two different ids for the menu links, so current sanity tests can access the V1 and new tests can target V2 ? 
It will be easier for tests, and having the same id twice on the page (Menu) is a regression.

**Comment by saulaski** [comment]
> However, this seems to be breaking all the sanity tests now, probably because the UI tests use the first link matching

I met the same issue recently. This is indeed Puppeteer's default behavior. A few months ago an improvement was proposed and rejected by Puppeteer's maintainers, because allowing multiple targets was considered bad testing practice.

We don't have to use 2 distinct identifiers, but 2 distinct CSS selectors. If somehow possible in current implementation, adding a distinct CSS class to both menu links might do the trick.

**Comment by jolelievre** [comment]
Hi @boubkerbribri,

the problem is that the ID of the link is automatically built based on the associated controller, since here both links re related to `AdminProducts` they have the same ID. Which is not a good thing of course, but it can be accepted temporarily I'd say.

Seems like sanity tests use the first matching link, whereas the breadcrumb builder uses the last one. So we are bound to have inconsistencies anyway, but I don't want to change the menu/breadcrumb code for this.

So for now I re-changed the tabs order so that the first one is the legacy one, thus it won't break the sanity tests at least. And for new UI tests I guess you can use a special selector to get the second link matching the ID.

So if it's alright for the QA testers that the link is changed for now, maybe it's an acceptable compromise?

**Comment by boubkerbribri** [comment]
I think it's the best we can hope for. So all good for me.

**Comment by jolelievre** [comment]
Ok, thanks @boubkerbribri I'm waiting for @Robin-Fischer-PS approval so we can re-approve the PR and maybe perform a last QA check for possible side effects

Keep in mind that this extra link is only here to ease our work temporarily and will be ultimately removed and replaced with a link to the new page (the legacy page will be completely removed)

**Comment by Robin-Fischer-PS** [comment]
Ok for me. We will indeed perform one last check before validating the PR :)

**Review by matthieu-rolland — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by HanaRebaiQA — APPROVED** [review]
Hello @jolelievre 

Thank you all for your feedback ! 
I checked again this PR. It seems good for me.
I checked with multistore, different languages (RTL and No RTL), responsive mode, different browsers (Chrome & Firefox)

**multistore :** 
image

**RTL language :** 
image

**Responsive mode :** 
image

**firefox :** 
image

The product of page v2 is well added on `product` table (a simple test example)  
https://watch.screencastify.com/v/ibRDdwBv9F17G9KVk0bJ

Uploading add product v2 on product table .mp4…



So, QA :heavy_check_mark: 

Thanks!
